packages/odoo/src/Adaptor.js

```diff
@@ -0,0 +1,186 @@
+import {
+  execute as commonExecute,
+  composeNextState,
+} from '@openfn/language-common';
+import { expandReferences } from '@openfn/language-common/util';
+
+import pkg from 'odoo-await';
+const Odoo = pkg;
+
+let odooConn = null;
+
+/**
+ * Execute a sequence of operations.
+ * Wraps `language-common/execute` to make working with this API easier.
+ * @example
+ * execute(
+ *   create('foo'),
+ *   delete('bar')
+ * )(state)
+ * @private
+ * @param {Operations} operations - Operations to be performed.
+ * @returns {Operation}
+ */
+export function execute(...operations) {
+  const initialState = {
+    references: [],
+    data: null,
+  };
+
+  return state => {
+    return commonExecute(login, ...operations)({ ...initialState, ...state });
+  };
+}
+
+/**
+ * Logs in to OpenSpp, gets a session token.
+ * @example
+ *  login(state)
+ * @private
+ * @param {State} state - Runtime state.
+ * @returns {State}
+ */
+async function login(state) {
+  const { baseUrl, username, password, database } = state.configuration;
+  odooConn = new Odoo({
+    baseUrl: baseUrl,
+    db: database,
+    username: username,
+    password: password,
+  });
+  try {
+    await odooConn.connect();
+  } catch (err) {
+    console.log(`✗ Error: ${err}`);
+    odooConn = null;
+  }
+  return state;
+}
+/**
+ * Create a record in Odoo
+ * @public
+ * @example
+ * create("res.partner", { name: "Kool Keith" });
+ * @function
+ * @param {string} model - The specific record model i.e. "res.partner"
+ * @param {object} data - The data to be created in JSON.
+ * @param {string} externalId - Optional external ID for the record.
```

Comment:
Maybe we can use an options object here, like:
```
create("res.partner", { name: "Kool Keith" }, { externalId: 22 );
```

The benefits are:
1) we can add more options later without a breaking change
2) the code is more readable when we explicitly pass in the `externalId` key. Otherwise it's just a random value in the signature and it looks a bit wierd